src/main/java/io/cryostat/util/RuleDeserializer.java

```diff
@@ -61,9 +61,57 @@ public Rule deserialize(JsonElement json, Type typeOf, JsonDeserializationContex
         JsonElement sanitized = JsonParser.parseString(Rule.sanitizeRuleName(dirty));
         jsonObject.add(name, sanitized); // replaces field with sanitized name
 
-        Rule rule = new Gson().fromJson(jsonObject, Rule.class);
+        Rule.Builder builder =
+                new Rule.Builder()
+                        .name(jsonObject.get(Rule.Attribute.NAME.getSerialKey()).getAsString())
+                        .targetAlias(
+                                jsonObject
+                                        .get(Rule.Attribute.TARGET_ALIAS.getSerialKey())
+                                        .getAsString())
+                        .description(
+                                jsonObject
+                                        .get(Rule.Attribute.DESCRIPTION.getSerialKey())
+                                        .getAsString())
+                        .eventSpecifier(
+                                jsonObject
+                                        .get(Rule.Attribute.EVENT_SPECIFIER.getSerialKey())
+                                        .getAsString());
 
-        rule.validate();
-        return rule;
+        builder = setOptionalInt(builder, Rule.Attribute.ARCHIVAL_PERIOD_SECONDS, jsonObject);
+        builder = setOptionalInt(builder, Rule.Attribute.PRESERVED_ARCHIVES, jsonObject);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_AGE_SECONDS, jsonObject);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_SIZE_BYTES, jsonObject);
+
+        return builder.build();
+    }
+
+    private static Rule.Builder setOptionalInt(
```

Comment:
Would it be useful if we refactored `setOptionalInt()` from `RulesPostHandler.java` to avoid duplicating code here? `setOptionalInt()` would need to handle both `RequestParameter`s or `JsonObject`s

Reply:
Yes, refactoring to avoid duplication would be very valuable. It isn't immediately obvious how to make it handle both `RequestParameters` and `JsonObject`s as a single method, but perhaps it can be done as an overloaded method (one signature for `RequestParameters` and another signature for `JsonObject`), which both call another common method that basically just contains the `switch` expression that is currently in the middle of the method body.

Reply:
Assuming the best place to put `setOptionalInt(JsonObject jsonObject)` is in `RulesPostHandler`, how can I make `setOptionalInt(JsonObject jsonObject)` visible to `RuleDeserializer` without making `RulesPostHandler` public?

Reply:
That probably wouldn't be the best place, actually - API handlers should ideally be pretty small and just deal with processing request stuff - grabbing values from path and query params, headers, etc. - and writing responses that are computed elsewhere. Any other deeper business logic should be refactored out for reuse, so that nothing has dependencies on API handlers (except maybe other API handlers when they're very closely related.

Reply:
How about in `Rule.java`? That would move the logic closer to `Builder` and `requireNonNegative`

Reply:
Sure, that sounds pretty reasonable

Reply:
`RequestParameters` is only visible in `api/v2/`. What's the best way to pass those parameters to `setOptionalInt()` in `Rule.java`?

Reply:
All you really need from the `RequestParameters` object is the form attributes, right? Maybe you can change the implementation in `Rule` to just take a `MultiMap` instead of the whole `RequestParameters`?

Reply:
Thanks! That looks better now.